File: iosApp/iosApp/Presentation/Settings/Notification/NotificationSettingsUiState.swift
//
//  NotificationSettingsUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/14.
//

import Foundation
import Shared

struct NotificationSettingsUiState {
    var isLoading: Bool
    var isSaving: Bool
    var errorMessage: String?
    var isCastRole: Bool
    var isPushNotificationsEnabled: Bool
    var isShiftNotificationsEnabled: Bool
    var isBirthdayNotificationsEnabled: Bool
    var isNoticeNotificationsEnabled: Bool
    var isFollowNotificationsEnabled: Bool
    var isEventNotificationsEnabled: Bool
    var isCommunityNotificationsEnabled: Bool
    var quietHoursOption: NotificationQuietHoursMode

    static let initial = NotificationSettingsUiState(
        isLoading: false,
        isSaving: false,
        errorMessage: nil,
        isCastRole: false,
        isPushNotificationsEnabled: true,
        isShiftNotificationsEnabled: true,
        isBirthdayNotificationsEnabled: true,
        isNoticeNotificationsEnabled: true,
        isFollowNotificationsEnabled: true,
        isEventNotificationsEnabled: true,
        isCommunityNotificationsEnabled: true,
        quietHoursOption: .off
    )
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/settings/notification/NotificationSettingsUiState.kt
package com.hhp227.concafe.presentation.settings.notification

import com.hhp227.concafe.domain.model.NotificationQuietHoursMode

data class NotificationSettingsUiState(
    val isLoading: Boolean,
    val isSaving: Boolean,
    val errorMessage: String?,
    val isCastRole: Boolean,
    val isPushNotificationsEnabled: Boolean,
    val isShiftNotificationsEnabled: Boolean,
    val isBirthdayNotificationsEnabled: Boolean,
    val isNoticeNotificationsEnabled: Boolean,
    val isFollowNotificationsEnabled: Boolean,
    val isEventNotificationsEnabled: Boolean,
    val isCommunityNotificationsEnabled: Boolean,
    val quietHoursOption: NotificationQuietHoursMode
) {
    companion object {
        fun initial(): NotificationSettingsUiState {
            return NotificationSettingsUiState(
                isLoading = false,
                isSaving = false,
                errorMessage = null,
                isCastRole = false,
                isPushNotificationsEnabled = true,
                isShiftNotificationsEnabled = true,
                isBirthdayNotificationsEnabled = true,
                isNoticeNotificationsEnabled = true,
                isFollowNotificationsEnabled = true,
                isEventNotificationsEnabled = true,
                isCommunityNotificationsEnabled = true,
                quietHoursOption = NotificationQuietHoursMode.OFF
            )
        }
    }
}
